test/new_relic/slack_notifications_tests/slack_notifier_tests.rb

```diff
@@ -0,0 +1,43 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require 'minitest/autorun'
+require_relative '../../../.github/workflows/scripts/slack_notifications/slack_notifier'
+
+class SlackNotifications < Minitest::Test
+  def test_send_slack_message_zero_args
+    assert_raises(ArgumentError) { SlackNotifier.send_slack_message() }
+  end
+
+  def test_send_slack_message_too_many_args
+    assert_raises(ArgumentError) {
+      SlackNotifier.send_slack_message("I am a notification message!", "But I'm one too many")
+    }
+  end
+
+  def test_send_slack_message
+    SlackNotifier.stub(:sleep, nil) do
+      HTTParty.stub(:post, nil) do
+        assert_nil SlackNotifier.send_slack_message("I am a notification message!")
+      end
+    end
+  end
+
+  def test_exit_handler_no_fail
```

Comment:
If you'd like to, the names for this test and the next should likely be updated to include `errors_array` since that's now the name of the tested method.

Also. since the `exit_handler` method has been broken into `errors_array` and `report_errors`, could you add tests for `report_errors` as well?

Reply:
Also, when running locally, I learned two interesting things: 


1. this test failed: 
```
  1) Failure:
SlackNotifications#test_exit_handler_no_fail [test/new_relic/slack_notifications_tests/slack_notifier_tests.rb:32]:
Expected [#<ArgumentError: wrong number of arguments (given 2, expected 0)>] to be empty.
```

2. The script files aren't getting included in SimpleCov. We might want to _try_ to adjust where we look so they get included.

Reply:
Thank you for this catch! Looks like an intermittent failure since I wasn't clearing out the `errors` array after adding a failure to it. For SimpleCov, it could be a fun exercise! Although I know the goal is to move all this functionality into its own gem, so not sure if its worth figuring out how to include them?

Reply:
Fixed naming and [added tests for `report_errors`](https://github.com/newrelic/newrelic-ruby-agent/pull/1432/files#diff-30d914ac7129feb2d13bf43e052cf29fe91564381ac3623355ac7f02ae77d096R58-R79).